zabbix/files/default/etc/zabbix/zabbix_agentd.conf.jinja

```diff
@@ -228,6 +228,35 @@ HostMetadata={{ settings.get('hostmetadata', defaults.hostmetadata) }}
 {% if settings.get('hostmetadataitem', defaults.get('hostmetadataitem', False)) -%}
 HostMetadataItem={{ settings.get('hostmetadataitem', defaults.hostmetadataitem) }}
 {% endif %}
+
+### Option: HostInterface
+#       Optional parameter that defines host interface.
+#       Host interface is used at host auto-registration process.
+#       An agent will issue an error and not start if the value is over limit of 255 characters.
+#       If not defined, value will be acquired from HostInterfaceItem.
+#
+# Mandatory: no
+# Range: 0-255 characters
+# Default:
+# HostInterface=
+{% if settings.get('hostinterface', defaults.get('hostinterface', False)) -%}
+HostInterface={{ settings.get('hostinterface', defaults.hostinterface) }}
+{% endif %}
+
+### Option: HostInterfaceItem
+#       Optional parameter that defines an item used for getting host interface.
+#       Host interface is used at host auto-registration process.
+#       During an auto-registration request an agent will log a warning message if
+#       the value returned by specified item is over limit of 255 characters.
+#       This option is only used when HostInterface is not defined.
+#
+# Mandatory: no
+# Default:
+# HostInterfaceItem=
+{% if settings.get('hostinterfaceitem', defaults.get('hostinterfaceitem', False)) -%}
+HostInterfaceItem={{ settings.get('hostinterfaceitem', defaults.hostinterfaceitem) }}
+{% endif %}
+
```

Comment:
```suggestion

{% if zabbix.version_repo|float >= 4.4 -%}
### Option: HostInterface
#       Optional parameter that defines host interface.
#       Host interface is used at host auto-registration process.
#       An agent will issue an error and not start if the value is over limit of 255 characters.
#       If not defined, value will be acquired from HostInterfaceItem.
#
# Mandatory: no
# Range: 0-255 characters
# Default:
# HostInterface=
{% if settings.get('hostinterface', defaults.get('hostinterface', False)) -%}
HostInterface={{ settings.get('hostinterface', defaults.hostinterface) }}
{% endif %}

### Option: HostInterfaceItem
#       Optional parameter that defines an item used for getting host interface.
#       Host interface is used at host auto-registration process.
#       During an auto-registration request an agent will log a warning message if
#       the value returned by specified item is over limit of 255 characters.
#       This option is only used when HostInterface is not defined.
#
# Mandatory: no
# Default:
# HostInterfaceItem=
{% if settings.get('hostinterfaceitem', defaults.get('hostinterfaceitem', False)) -%}
HostInterfaceItem={{ settings.get('hostinterfaceitem', defaults.hostinterfaceitem) }}
{% endif %}
{% endif -%}

```

Also, those parameters only supported since Zabbix Agent version 4.4.0 I can suggest to put them inside `if` block, so old agents won't get unsupported parameters in their configuration. Sorry not to mention this earlier.

Reply:
It's a Jinja thing, "whitespace control", minus at the end of block will remove newline which otherwise will be added to rendered file.
More info https://jinja.palletsprojects.com/en/2.11.x/templates/#whitespace-control